src/fidasim.f90

```diff
@@ -10155,7 +10162,11 @@ subroutine neutron_f
                             call get_neutron_rate(plasma, erel, rate)
                             if(inputs%calc_neutron.ge.2) then
                                 neutron%weight(ie,ip,ir,iz,iphi) = neutron%weight(ie,ip,ir,iz,iphi) &
-                                                                 + rate*factor
+                                                                 + rate * factor
+                                !!! Double check factor
+                                neutron%emis(ir,iz,iphi) = neutron%emis(ir,iz,iphi) &
```

Comment:
You will put the emissivity calculation in an OMP critical block

Reply:
Why should the emissivity calculation be done with one node at a time?

Reply:
You are writing to the same memory location across multiple OpenMP threads. In order to prevent one thread from overwriting another you have to indicate to OpenMP that only one thread can operate on this section at a time.

Reply:
Why doesn't the neutron weight have the same issue?

Reply:
The calculation loops over the energy,pitch,r,z,and phi. Every loop is accessing a different memory location so there is no chance of one thread overwriting another. However, for the emissivity the the array only accesses r/z/phi so that means multiple values of energy and pitch are writing to the same memory location.

Reply:
Ah! That makes sense now. Thanks.